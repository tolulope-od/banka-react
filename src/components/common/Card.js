import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ cardContent }) => {
  return <div className="account-stats-card">{cardContent}</div>;
};

Card.propTypes = {
  cardContent: PropTypes.object.isRequired
};

export default Card;
