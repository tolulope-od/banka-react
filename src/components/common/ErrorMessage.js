import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({ errors }) => (
  <div>
    <img
      src="https://res.cloudinary.com/tolulope-od/image/upload/v1563463864/action-error_ohkpfa.png"
      alt="error-image"
      width="80"
    />
    <h3>{errors.error || 'Something Unusual happened'}</h3>
    <input
      type="submit"
      className="accnt-create-btn"
      name="addAccount"
      value="OK"
      id="submit-btn"
    />
  </div>
);

ErrorMessage.propTypes = {
  errors: PropTypes.object
};

export default ErrorMessage;
