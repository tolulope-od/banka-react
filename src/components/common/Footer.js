import React from 'react';

const Footer = () => {
  return (
    <footer className="footer">
      <p className="footerTxt center-text">Banka Inc &copy; 2019</p>
    </footer>
  );
};

export default Footer;
