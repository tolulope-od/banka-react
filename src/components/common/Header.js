import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { logout } from '../../redux/actions/authActions';
import avatar from '../../images/avatar.png';

export class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false
    };
    this.onClick = this.onClick.bind(this);
    this.onLogoutClick = this.onLogoutClick.bind(this);
  }

  onClick() {
    this.setState({ isOpen: !this.state.isOpen });
  }

  onLogoutClick() {
    this.props.logout();
  }

  render() {
    const { isAuthenticated, user } = this.props.auth;
    const { pathname } = this.props.history.location;
    const { isOpen } = this.state;
    return isAuthenticated ? (
      <div>
        <div id="navbar" className="menu-nav">
          <Link to="/dashboard" id="logo">
            <i className="fas fa-coins"></i>Banka
          </Link>
          <div id="navbar-right">
            <div className="profile-action" id="drop-down" onClick={this.onClick}>
              <img src={avatar} alt="Profile Picture" className="avatar-img" />
              <i className="fas fa-chevron-down"></i>
            </div>
          </div>
        </div>
        <div
          className={isOpen ? 'profile-dropdown-active' : 'profile-dropdown'}
          id="dropdown-contents"
        >
          <Link
            to="/dashboard"
            className="dropdown-item profile"
            id={pathname.includes('dashboard') ? 'active' : 'inactive'}
          >
            <i className="fas fa-columns icon"></i> Dashboard
          </Link>

          <Link
            to="/viewaccounts"
            className="dropdown-item profile"
            id={pathname.includes('account') ? 'active' : 'inactive'}
          >
            <i className="fas fa-file-invoice-dollar icon"></i> Accounts
          </Link>
          {user.type === 'client' ? null : (
            <Link
              to="/users"
              className="dropdown-item profile"
              id={pathname.includes('user') ? 'active' : 'inactive'}
            >
              <i className="fas fa-users icon"></i> Users
            </Link>
          )}
          <Link
            to="/profile"
            className="dropdown-item profile"
            id={pathname.includes('profile') ? 'active' : 'inactive'}
          >
            <i className="fas fa-user-circle icon"></i> Profile
          </Link>
          <Link to="/" className="dropdown-item" id="logout-btn" onClick={this.onLogoutClick}>
            <i className="fas fa-sign-out-alt icon"></i>Logout
          </Link>
        </div>
      </div>
    ) : (
      <div id="navbar">
        <Link to="/" id="logo">
          <i className="fas fa-coins"></i>Banka
        </Link>
        <div id="navbar-right">
          <Link to="/login" className="login-btn">
            Login
          </Link>
        </div>
      </div>
    );
  }
}

Header.propTypes = {
  auth: PropTypes.object.isRequired,
  logout: PropTypes.func.isRequired,
  user: PropTypes.object,
  history: PropTypes.object
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { logout }
)(Header);
