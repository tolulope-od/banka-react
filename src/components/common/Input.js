import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ type, name, placeholder, id, value, onChange, required, disabled }) => {
  return (
    <input
      type={type}
      name={name}
      placeholder={placeholder}
      id={id}
      value={value}
      onChange={onChange}
      required={required}
      disabled={disabled}
    />
  );
};

Input.propTypes = {
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  required: PropTypes.bool,
  disconnect: PropTypes.bool
};

export default Input;
