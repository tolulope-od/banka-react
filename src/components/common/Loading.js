import React from 'react';

const Loading = () => {
  return (
    <center>
      <img
        src="https://res.cloudinary.com/tolulope-od/image/upload/v1557908999/loading_o1y5v6.gif"
        width="150"
      />
    </center>
  );
};

export default Loading;
