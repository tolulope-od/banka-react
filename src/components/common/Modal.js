import React from 'react';
import PropTypes from 'prop-types';

const Modal = ({ onClick, modalContent, className }) => {
  return (
    <div id="myModal" className={className} onClick={onClick}>
      <div className="modal-content" id="account-modal-content" onClick={onClick}>
        <span className="close" onClick={onClick} id="close-button">
          &times;
        </span>
        <div>{modalContent}</div>
      </div>
    </div>
  );
};

Modal.propTypes = {
  onClick: PropTypes.func.isRequired,
  modalContent: PropTypes.object.isRequired,
  className: PropTypes.string.isRequired
};

export default Modal;
