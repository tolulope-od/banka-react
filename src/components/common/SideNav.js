/* eslint-disable no-nested-ternary */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export class SideNav extends Component {
  render() {
    const { isAuthenticated, user } = this.props.auth;
    const { pathname } = this.props.history.location;
    const label = pathname.includes('dashboard')
      ? 'DASHBOARD'
      : pathname.includes('account')
      ? 'ACCOUNT'
      : pathname.includes('profile')
      ? 'PROFILE'
      : 'USERS';
    return isAuthenticated ? (
      <div>
        <div className="item-content-container">
          <div className="side-nav">
            <div className="side-nav-content">
              <h5 className={user.type === 'staff' ? 'side-nav-greet-admin' : 'side-nav-greet'}>
                Welcome,
              </h5>
              <h4 className="side-nav-name" id="side-nav-name">
                {user.firstName}
              </h4>
              <Link
                to="/dashboard"
                className="side-link"
                id={pathname.includes('dashboard') ? 'sidelink-active' : null}
              >
                <i className="fas fa-columns"></i> Dashboard
              </Link>
              <Link
                to="/viewaccounts"
                className="side-link"
                id={pathname.includes('account') ? 'sidelink-active' : null}
              >
                <i className="fas fa-file-invoice-dollar"></i> Accounts
              </Link>
              {user.type === 'staff' ? (
                <Link
                  to="/users"
                  className="side-link"
                  id={pathname.includes('user') ? 'sidelink-active' : null}
                >
                  <i className="fas fa-users"></i> Users
                </Link>
              ) : null}
              <Link
                to="/profile"
                className="side-link"
                id={pathname.includes('profile') ? 'sidelink-active' : null}
              >
                <i className="fas fa-user-circle"></i> Profile
              </Link>
            </div>
          </div>
          <div className="item-content-container dashboard">
            <div className="col s5">
              <h2 className="label-lg">
                <strong>{label}</strong>
              </h2>
            </div>
          </div>
        </div>
        <div className="nav-area"></div>
      </div>
    ) : null;
  }
}

SideNav.propTypes = {
  auth: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  null
)(SideNav);
