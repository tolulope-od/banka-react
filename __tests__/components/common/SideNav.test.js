import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SideNav } from '../../../src/components/common/SideNav';

const shallowSetup = () => {
  const props = {
    auth: {
      isAuthenticated: true,
      user: {
        firstName: 'Obiwan',
        lastName: 'Kenobi',
        type: 'client'
      }
    },
    history: { location: { pathname: '' } }
  };

  const enzymeWrapper = shallow(<SideNav {...props} />);

  return { props, enzymeWrapper };
};

describe('<SideNav', () => {
  const { props } = shallowSetup();
  it('Renders the Header component', () => {
    const wrapper = shallow(<SideNav {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should return an active id on the dashboard icon it is active', () => {
    const wrapper = shallow(<SideNav {...props} />);
    wrapper.setProps({
      history: { location: { pathname: 'dashboard' } }
    });
    expect(wrapper.find('#sidelink-active').length).toEqual(1);
  });

  it('Should return an active id on the account icon it is active', () => {
    const wrapper = shallow(<SideNav {...props} />);
    wrapper.setProps({
      history: { location: { pathname: 'viewaccounts' } }
    });
    expect(wrapper.find('#sidelink-active').length).toEqual(1);
  });

  it('Should return an active id on the profile icon it is active', () => {
    const wrapper = shallow(<SideNav {...props} />);
    wrapper.setProps({
      history: { location: { pathname: 'profile' } }
    });
    expect(wrapper.find('#sidelink-active').length).toEqual(1);
  });

  it('Should render a different element className for an admin user', () => {
    const wrapper = shallow(<SideNav {...props} />);
    wrapper.setProps({
      auth: { isAuthenticated: true, user: { type: 'staff' } }
    });
    expect(wrapper.find('.side-nav-greet-admin').length).toEqual(1);
  });

  it('Should return an active id on the user icon it is active', () => {
    const wrapper = shallow(<SideNav {...props} />);
    wrapper.setProps({
      auth: { isAuthenticated: true, user: { type: 'staff' } },
      history: { location: { pathname: 'users' } }
    });
    expect(wrapper.find('#sidelink-active').length).toEqual(1);
  });

  it('Should not render a menu when the user is not authenticated', () => {
    const wrapper = shallow(<SideNav {...props} />);
    wrapper.setProps({
      auth: { isAuthenticated: false, user: {} },
      history: { location: { pathname: '' } }
    });
    expect(wrapper.find('#sidelink-active').length).toEqual(0);
  });
});
