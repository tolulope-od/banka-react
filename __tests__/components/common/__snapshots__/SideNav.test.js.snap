// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav Renders the Header component 1`] = `
<div
  className="side-nav"
>
  <div
    className="side-nav-content"
  >
    <h5
      className="side-nav-greet"
    >
      Welcome,
    </h5>
    <h4
      className="side-nav-name"
      id="side-nav-name"
    >
      Obiwan
    </h4>
    <Link
      className="side-link"
      id={null}
      to="/dashboard"
    >
      <i
        className="fas fa-columns"
      />
       Dashboard
    </Link>
    <Link
      className="side-link"
      id={null}
      to="/viewaccounts"
    >
      <i
        className="fas fa-file-invoice-dollar"
      />
       Accounts
    </Link>
    <Link
      className="side-link"
      id={null}
      to="/profile"
    >
      <i
        className="fas fa-user-circle"
      />
       Profile
    </Link>
  </div>
</div>
`;
